lib/util/http/host-rules.ts

```diff
@@ -14,7 +14,7 @@ export function applyHostRules(url: string, inOptions: GotOptions): GotOptions {
   const { username, password, token, enabled } = foundRules;
   if (options.headers?.authorization || options.password || options.token) {
     logger.trace('Authorization already set for host: ' + options.hostname);
-  } else if (password) {
+  } else if (password || username) {
```

Comment:
Needed to add this as well to enable "passwordless" basic auth just using a token as username

Reply:
TODO: Sanitize `username` from logs if no `password` key is present

Reply:
Done here: https://github.com/renovatebot/renovate/pull/6869/commits/503b9a36f0f7da49cd090e149055700d30c0e533 but seems pretty ugly to me. Any suggestions?